woocommerce/woocommerce-ios
# Blaze: Add payment info screen

Part of #11702 


## Description

This PR adds a new screen for payment info after confirming details for a Blaze campaign. The view receives campaign details from the creation form and lets the user select a payment method before proceeding to create a campaign.

Changes:
- Added `CreditCardType` for displaying card types on the UI. We are reusing the same images as WCPay for simplicity - note that the icon for the Visa card is not the same as on the design. I hope it's ok.
- Added navigation from the campaign creation form to the payment info screen.
- Added view and view model for the payment info screen. This includes only fetching and displaying card details for now. Due to the size of this PR, I'll add error handling and unit tests in a separate PR.

## Testing instructions

- Log in to a store that's eligible for Blaze.
- If the campaign creation entry point is available on the Dashboard screen, select it and select a product if needed. Otherwise, navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- Confirm that the payment info screen works correctly. 
- Comment out the code to keep `isFetchingPaymentInfo` as `true` and `selectedPaymentMethod` as nil. The activity indicator should be displayed for the loading state.

## Screenshots



Loading state | Result state
--- | --- 
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/c389f474-7233-4171-8d4f-af2a9eae2d72" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/046e70c7-7860-42d6-9378-4d11fba8d5d7" width=320 />


---
